### ISSUE TITLE
Can't compose within a class if it shares names with a global class

### ISSUE THREAD

**Issue by kevinkace** [tivac/modular-css#192]
Similar to #184

Works:
```css
.b { color: red; }
:global(.b) { color: blue; }
```

Doesn't work:
```css
.a { background: orange; }
.b { composes: a; color: red; }
:global(.b) { color: blue; }
```

**Comment by tivac** [comment]
This is happening because inside the scoping plugin we build up a map of original class/id to renamed class/id.

For the broken example, that map looks like this:

```js
{
    a : [ "abc123_a" ], // transformed
    b : [ "b" ]         // not transformed due to :global() wrapping
}
```

because the `:global(.b)` comes after the original `.b` declaration and overwrites it. I see two options here. Will post as two separate comments so reactions can be used to cast votes.

**Comment by tivac** [comment]
## Duplicated identifiers in a file throw an error (easy)

- :+1: Simple
- :+1: Output object is always easy to reference because there can't be duplicates
- :-1: Requires more thought on naming

So your example of 

```css
.a { background: orange; }
.b { composes: a; color: red; }
:global(.b) { color: blue; }
```

Would throw an error like "Error: unable to re-use selector '.b'"

**Comment by tivac** [comment]
## `:global()` identifiers are exposed differently (hard)

In JS you'd end up with something like this:

```js
{
    a : "abc123_a",
    b : "abc123_a abc123_b",

    // One option for exposing globals in their own namespace
    global : {
        b : "b"
    }

    // Another option would be to prefix the keys in the object
    global_b : "b"
}
```

- :+1: You can mix local/global identifiers w/o concerns
- :-1: Referencing a global identifier from the exported object is different from a local one

**Comment by tivac** [comment]
Fixed by 2ca52ad277d52ee7162c126009fd59c6c7f9c087
